apis/management.cattle.io/v3/pipeline_types.go

```diff
@@ -1,10 +1,19 @@
 package v3
 
 import (
+	"github.com/rancher/norman/condition"
 	"github.com/rancher/norman/types"
+	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type PipelineConditionType string
+
+const (
+	PipelineExecutionConditonProvisioned condition.Cond = "Provisioned"
+	PipelineExecutionConditionFailed     condition.Cond = "Failed"
```

Comment:
this is a bit vague for a name. Any condition can fail, I would not recommend to use it for the name. Plus condition should be "positive", meaning that when it's value is true, everything is good. Otherwise state aggregation on api level won't work by default.

What does this condition mean? Is it an equivalent of Provisioned=false? Or is it more about execution itself? If the latter, can we change it to be `PipelineExecutionConditionUpdated` or `PipelineExecutionConditionExecuted`

Reply:
It means the pipelineexecution failed the build, for example got non-zero exit of the scripts. So is it good to be ConditionSucceed, updated/executed is not what it supposed to be.

Reply:
@gitlawr PipelineExecutionConditionSucceed or Finished is better, as it falls under rule that when set to `True`, it's a good thing